### PR TITLE
Example-App-Base: Enable strictNullChecks

### PR DESCRIPTION
**Pull request by christiango**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes
Enable strictNullChecks for example app base. Fix implicit anys.

### CODE REVIEW

File: packages/example-app-base/src/components/Highlight/Highlight.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import * as React from 'react';
-import * as hljs from 'highlight.js/lib/highlight';
+import { registerLanguage } from 'highlight.js';
 import * as javascript from 'highlight.js/lib/languages/javascript';
```
Comment (christiango) [inline packages/example-app-base/src/components/Highlight/Highlight.tsx]: This import here is preventing us from turning on noImplicitAny. Not familiar enough with highlight.js to figure out if there is another way to do the initialization using stuff in the @types.

File: packages/example-app-base/src/utilities/data.ts [hunk 2]
```diff
@@ -30,11 +30,12 @@ export function createListItems(count: number, startIndex = 0): any {
 
 export function createGroups(
   groupCount: number, groupDepth: number, startIndex: number,
-  itemsPerGroup: number, level?: number, key?: string) {
-  key = key ? key + '-' : '';
-  level = level ? level : 0;
+  itemsPerGroup: number, level: number = 0, key: string = '') {
+  if (key !== '') {
```
Comment (christiango) [inline packages/example-app-base/src/utilities/data.ts]: This change here wasn't mechanical, would appreciate another set of eyes looking at this closely to make sure I didn't mess it up.

File: packages/example-app-base/package.json [hunk 3]
```diff
@@ -20,6 +20,7 @@
     "@types/chai": "3.4.35",
     "@types/enzyme": "2.8.0",
     "@types/es6-promise": "^0.0.32",
+    "@types/highlight.js": "^9.1.9",
```
Comment (dzearing) [inline packages/example-app-base/package.json]: Can you remove the carets from these? What we've seen is that increasing the version can break the build unexpectedly, especially in cases like the react typings, which depend on the latest typescript. So I've removed the carets to ensure that we don't automatically pick up breaking changes for typings.

File: packages/example-app-base/src/utilities/data.ts (hunk 2 above)
Comment (dzearing) [inline packages/example-app-base/src/utilities/data.ts]: looks fine.

**Pull request merged**